### PR TITLE
Naming improvements

### PR DESCRIPTION
**Pull request by arielgabizon**
- `ZCProof` is a too general name, now that we also have `GrothProof` used in sprout proofs.
So I changed the name of this object to `PHGRProof`.

- In some files `pubKeyHash` was used as a var name, whereas it wasn't the pubkey hash,
but the pubkey itself. So I changed the var name to `joinSplitPubKey`

### CODE REVIEW

**Comment by daira** [comment]
There is no need to declare parameters *of value type* (e.g. ``uint64_t``) as ``const``; they're passed by value so there is no possibility of affecting the caller's copy. My merge undid those changes (maybe I should have done that as another commit).

File: src/zcash/JoinSplit.cpp [hunk 1]
```diff
@@ -137,16 +137,18 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
         } catch (...) {
             return false;
         }
-    }
+    
+}
```
Comment (daira) [inline src/zcash/JoinSplit.cpp]: Undo this change and then squash, please.

Comment (arielgabizon) [reply src/zcash/JoinSplit.cpp]: Should I keep your merge commit, or remove the `const`s in my commit and force push?

Comment (arielgabizon) [reply src/zcash/JoinSplit.cpp]: Squash how far? Should I also squash your merge commit?

Comment (daira) [reply src/zcash/JoinSplit.cpp]: Yes.

**Comment by arielgabizon** [comment]
@daira addressed your comment. Got into a git mess, so just repulled from master and redid the changed :)

**Comment by arielgabizon** [comment]
Ohh oops, maybe I just redid what you did :(

**Comment by daira** [comment]
Yes you did. It's okay, I'll rereview now.

**Comment by str4d** [comment]
@zkbot r+
